			     BASH PATCH REPORT
			     =================

Bash-Release:	5.1
Patch-ID:	bash51-011

Bug-Reported-by:	Alex fxmbsw7 Ratchev <fxmbsw7@gmail.com>
Bug-Reference-ID:	<CAALKErFrm+yuwy=ioK+RmjCiwxTDmzeMFWqLF2BEWYtkGS5UFQ@mail.gmail.com>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2020-11/msg00064.html

Bug-Description:

When reading a compound assignment, and running it through the parser to
split it into words, we need to save and restore any alias we're currently
expanding.

--- a/parse.y
+++ b/parse.y
@@ -6493,10 +6493,8 @@ parse_string_to_word_list (s, flags, who
   old_expand_aliases = expand_aliases;
 
   push_stream (1);
-#if 0 /* TAG: bash-5.2 Alex fxmbsw7 Ratchev <fxmbsw7@gmail.com> 11/17/2020 */
   if (ea = expanding_alias ())
     parser_save_alias ();
-#endif
   last_read_token = WORD;		/* WORD to allow reserved words here */
   current_command_line_count = 0;
   echo_input_at_read = expand_aliases = 0;
@@ -6531,10 +6529,8 @@ parse_string_to_word_list (s, flags, who
   last_read_token = '\n';
   pop_stream ();
 
-#if 0 /* TAG: bash-5.2 */
   if (ea)
     parser_restore_alias ();
-#endif
 
 #if defined (HISTORY)
   remember_on_history = old_remember_on_history;
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 10
+#define PATCHLEVEL 11
 
 #endif /* _PATCHLEVEL_H_ */
--- a/y.tab.c
+++ b/y.tab.c
@@ -8787,10 +8787,8 @@ parse_string_to_word_list (s, flags, who
   old_expand_aliases = expand_aliases;
 
   push_stream (1);
-#if 0 /* TAG: bash-5.2 Alex fxmbsw7 Ratchev <fxmbsw7@gmail.com> 11/17/2020 */
   if (ea = expanding_alias ())
     parser_save_alias ();
-#endif
   last_read_token = WORD;		/* WORD to allow reserved words here */
   current_command_line_count = 0;
   echo_input_at_read = expand_aliases = 0;
@@ -8825,10 +8823,8 @@ parse_string_to_word_list (s, flags, who
   last_read_token = '\n';
   pop_stream ();
 
-#if 0 /* TAG: bash-5.2 */
   if (ea)
     parser_restore_alias ();
-#endif
 
 #if defined (HISTORY)
   remember_on_history = old_remember_on_history;
